test/fuse/linux/setstat_test.cc

```diff
@@ -0,0 +1,358 @@
+// Copyright 2020 The gVisor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <errno.h>
+#include <fcntl.h>
+#include <linux/fuse.h>
+#include <sys/stat.h>
+#include <sys/statfs.h>
+#include <sys/types.h>
+#include <unistd.h>
+#include <utime.h>
+
+#include <string>
+#include <vector>
+
+#include "gtest/gtest.h"
+#include "test/fuse/linux/fuse_base.h"
+#include "test/util/fs_util.h"
+#include "test/util/fuse_util.h"
+#include "test/util/test_util.h"
+
+namespace gvisor {
+namespace testing {
+
+namespace {
+
+class SetStatTest : public FuseTest {
+ public:
+  SetStatTest()
+      : test_dir_("testdir"),
+        test_file_("testfile"),
+        test_dir_mode_(S_IFDIR | S_IRUSR | S_IWUSR | S_IXUSR),
+        test_file_mode_(S_IFREG | S_IRUSR | S_IWUSR | S_IXUSR) {}
+
+  void SetUp() override {
+    FuseTest::SetUp();
+    test_dir_path_ = JoinPath(mount_point_.path(), test_dir_);
+    test_file_path_ = JoinPath(mount_point_.path(), test_file_);
+  }
+  void TearDown() override {
```

Comment:
better comment why we want to skip unconsumed request, plus we can always do `SkipServerActualRequest()` so not sure if this override is needed (if so, probably better to also explain it in comment)

Reply:
Use close and SkipServerActualRequest instead.

Reply:
After `FuseFdTest` is added in #9, I use `CloseFD()` to ensure the fd will be closed and the request will be consumed. No need to override `TearDown`.